projects/observability/src/shared/components/topology/d3/layouts/graph-layout.ts

```diff
@@ -0,0 +1,76 @@
+import { RenderableTopology, RenderableTopologyNode, TopologyCoordinates, TopologyEdge, TopologyNode } from "../../topology";
+
+export class GraphLayout {
+  public layout(topology: RenderableTopology<TopologyNode, TopologyEdge>): void {
+
+    const levelToNodesMap = new Map<number, RenderableTopologyNode[]>();
+    const nodeToLevelMap = new Map<RenderableTopologyNode, number>();
+    const nodeCoOrdinatesMap: Map<string, TopologyCoordinates> = new Map<string, TopologyCoordinates>();
```

Comment:
can these be on the class level instead of the function level? As I assume, there will be only one instance of that per layout class.

Reply:
If we want to layout again we can probably clear the maps and refill them again

Reply:
Yeah right, will update that

Reply:
updated